### PR TITLE
Profile type validation and better icon handle

### PR DESCRIPTION
**Pull request by KiritoDv**
* Using icons from the profiles instead of the config

### CODE REVIEW

File: SetupContextMenu.ps1 [hunk 1]
```diff
@@ -111,27 +124,28 @@ $profiles | ForEach-Object {
         
         [void](New-ItemProperty -Path $subItemCMDPath -Name "(default)" -PropertyType String -Value "$env:LOCALAPPDATA\Microsoft\WindowsApps\wt.exe -p `"$profileName`" -d `"%V`"")
 
-        if ($configEntry.icon) {
-            $icoPath = $configEntry.icon
+
+        if ($_.icon) {
+            $icoPath = $_.icon
         }
```
Comment (kerol2r20) [inline SetupContextMenu.ps1]: I think this rule should be reserved.

Maybe you could prepend the following
```
if($_.icon)
```

File: SetupContextMenu.ps1 [hunk 2]
```diff
@@ -111,27 +124,28 @@ $profiles | ForEach-Object {
...
         elseif(($commandLine -match "^cmd\.exe\s?.*")) {
-            $icoPath = $cmdIcoFileName
+            $icoPath = "$resourcePath$cmdIcoFileName"
```
Comment (kerol2r20) [inline SetupContextMenu.ps1]: I wonder why you want to set `$resourcePath` here.
This is the second PR point out this change 🤣

**Comment by kerol2r20** [comment]
Nice exception handling
Thanks~

**Comment by KiritoDv** [comment]
Check the latest commit, i add support for config and profiles icon

**Pull request merged**